Feature/google pay alternative flow

avargic opened the pull request:
- Add additional alternative flow to sdk for google pay - less implementation is required from the client app, but is less flexible (e.g. if shipping is required)
- The client simply creates an instance of the GooglePayHandler and calls startGooglePayPayment for the payment session
- The client implements GooglePayHandlerCallbacks to receive information/results for the google pay flow
- GooglePayHandler internally handles the fetching of the paymentsClient, creating the googlePayButton, and displaying the payment method sheet and calling confirmPayment once the button is clicked

avargic committed:
- add alternative google pay flow to sdk
- fix MR comments

avargic committed:
- move strings to resources file

avargic committed:
- monri sdk version bump

avargic committed:
- target api version bump
- fix edge-to-edge display for confirmPayment activity

avargic committed:
- fix build tools and compileSdkVersion

luka-dragutin-asee commented on app/src/main/java/com/monri/android/example/GooglePayActivity.java:
```diff
@@ -55,6 +54,14 @@ public class GooglePayActivity extends AppCompatActivity implements ViewDelegate
     private MonriGooglePaymentRequestHelper monriGooglePaymentRequestHelper;
     private final int apiVersion = 2;
     private final int apiVersionMinor = 0;
+    final CustomerParams testCustomerParams = new CustomerParams()
```
private

luka-dragutin-asee commented on app/src/main/java/com/monri/android/example/GooglePayActivitySimple.java:
```diff
@@ -0,0 +1,182 @@
+    }
+
+    @Override
+    protected void onCreate(@Nullable Bundle savedInstanceState) {
```
final

luka-dragutin-asee commented on app/src/main/java/com/monri/android/example/GooglePayActivitySimple.java:
```diff
@@ -0,0 +1,182 @@
+    }
+
+    @Override
+    public void onConfirmPaymentResult(PaymentResult result, Throwable cause) {
```
final

luka-dragutin-asee commented on app/src/main/java/com/monri/android/example/OrderRepository.java:
```diff
@@ -50,6 +50,8 @@ public class OrderRepository {
 
     private static final String CONTENT_TYPE = "application/json";
 
+    static final String CREATE_PAYMENT_SESSION_APPROVED = "approved";
```
Gdje se ovo koristi?

avargic replied:
u drugim klasama za provjeru jel succesful response, prebaceno u enum u Status model

luka-dragutin-asee commented on monri/src/main/java/com/monri/android/MonriApiImpl.java:
```diff
@@ -26,7 +26,7 @@ class MonriApiImpl implements MonriApi {
         taskRunner.executeAsync(
                 () -> {
-                    MonriHttpResult<JSONObject> result = monriHttpApi.startGooglePayPayment(paymentId);
+                    final MonriHttpResult<JSONObject> result = monriHttpApi.startGooglePayPayment(paymentId);
```
Bravo

luka-dragutin-asee commented on monri/src/main/java/com/monri/android/google_pay/GooglePayHandlerCallbacks.java:
```diff
@@ -0,0 +1,17 @@
+
+    void onNotReadyToPayWithGooglePay();
+
+    void onGooglePayButtonReady(final PayButton payButton);
```
ode ne ide final, samo u implementaciju (metodu el)

luka-dragutin-asee commented on monri/src/main/java/com/monri/android/google_pay/GooglePayHandlerCallbacks.java:
```diff
@@ -0,0 +1,17 @@
+
+    void onConfirmPaymentResult(final PaymentResult result, final Throwable cause);
+
+    void onError(final GooglePayHandlerException googlePayHandlerException);
```
isto kao gore

luka-dragutin-asee commented on monri/src/main/java/com/monri/android/google_pay/GooglePayHandler.java:
```diff
@@ -0,0 +1,161 @@
+            googlePayHandlerCallbacks.onError(new GooglePayHandlerException(PARSE_PAYMENT_METHOD_DATA_JSON_ERROR));
+        }
+
+        confirmGooglePayPayment(googlePaymentMethodData);
```
jel se ovo uvijek mora pozvati? i kad se dogodi exception?

luka-dragutin-asee commented on monri/src/main/java/com/monri/android/google_pay/GooglePayHandler.java:
```diff
@@ -0,0 +1,161 @@
+                                .addOnFailureListener(e ->
+                                                              googlePayHandlerCallbacks.onError(
+                                                                      new GooglePayHandlerException(IS_READY_TO_PAY_PAYMENTS_CLIENT_ERROR)
+                                                              ));
```
Jel ovo izgleda u IDE-u ovako ili samo github luduje?

avargic committed:
- fix final modifiers
- remove apiVersion and apiVersionMinor from MonriGooglePaymentRequestHelper constructor
- remove unused GooglePayHandler exception
- add Status model
- gradle version bump
